### PR TITLE
tsdb.BeyondTimeRetention: Fix comment and test at retention duration

### PR DESCRIPTION
**Pull request by aknuds1**
I noticed that the comment in `tsdb.BeyondTimeRetention` regarding which blocks to delete is out of date since #9633. I'm proposing to make it reflect that blocks with max time difference, versus the first block, greater than _or equal to_ the retention duration are deleted.

I also propose to make the test `TestTimeRetention` contain also a test case for when a block is exactly at the retention duration (which is what #9633 changes). Currently that particular case is untested. On suggestion from @machine424, remove `TestDB_BeyondTimeRetention` since it overlaps with `TestTimeRetention`.

cc @darshanime

### CODE REVIEW

**Review by beorn7 — APPROVED** [review]
I would like to get an approval from @jesusvazquez, too, before merging.

**Comment by machine424** [comment]
Nice catch @aknuds1, The test doesn't check the `=` case. For the reset, we already have [TestTimeRetention](https://github.com/prometheus/prometheus/blob/11fc7b1d835e04a7f426841fd0cc8ab178d12717/tsdb/db_test.go#L1493), maybe we can go further and add `TestDB_BeyondTimeRetention` as a case in there (merge the two tests).

**Comment by aknuds1** [comment]
I didn't see `TestTimeRetention` @machine424, but sounds like a good idea.

**Comment by aknuds1** [comment]
I'll have a look at `TestTimeRetentation` tomorrow @machine424, and consider merging the tests.

**Comment by aknuds1** [comment]
@machine424 I merged `TestDB_BeyondTimeRetention` into `TestTimeRetention`, just making sure that the case of blocks exactly on the retention boundary is covered. PTAL.

File: tsdb/db_test.go
Comment (machine424) [inline tsdb/db_test.go:1473]: I'm afraid, we no longer test the `>` case now.
I was thinking about having two testCases in here, with the existing

```
[]*BlockMeta{
{MinTime: 500, MaxTime: 900}, // Oldest block
{MinTime: 1000, MaxTime: 1500},
{MinTime: 1500, MaxTime: 2000},
}
```

to test the `>` case.

And add

```
[]*BlockMeta{
{MinTime: 500, MaxTime: 1000}, // Oldest block
{MinTime: 1000, MaxTime: 1500},
{MinTime: 1500, MaxTime: 2000},
}
```

to test the `=` case.

Comment (aknuds1) [reply tsdb/db_test.go:1473]: Sounds good, on it.

Comment (machine424) [reply tsdb/db_test.go:1473]: (the existing checks should pass for both of them without any tweaks, we just need to add the loop)

Comment (aknuds1) [reply tsdb/db_test.go]: @machine424 I split `TestTimeRetention` into two sub-tests, one for each case (I verified that they break if I modify the retention check). PTAL.

**Review by bwplotka — APPROVED** [review]
(Hello from the bug scrub meeting!)

LGTM, let's merge this in 👍🏽

**Pull request merged**
